pkg/model/application_test.go

```diff
@@ -97,3 +98,62 @@ func TestApplication_ContainLabels(t *testing.T) {
 		})
 	}
 }
+
+func TestToApplicationKind(t *testing.T) {
+	testcases := []struct {
+		name         string
+		kind         string
+		expectedKind ApplicationKind
+		valid        bool
+	}{
+		{
+			name:         "KubernetesApp passed",
+			kind:         "KubernetesApp",
+			expectedKind: -1,
+			valid:        false,
+		},
+		{
+			name:         "Kubernetes passed",
+			kind:         "Kubernetes",
+			expectedKind: ApplicationKind_KUBERNETES,
+			valid:        true,
+		},
+		{
+			name:         "Terraform passed",
+			kind:         "Terraform",
+			expectedKind: ApplicationKind_TERRAFORM,
+			valid:        true,
+		},
+		{
+			name:         "Cloudrun passed",
+			kind:         "Cloudrun",
+			expectedKind: ApplicationKind_CLOUDRUN,
+			valid:        true,
+		},
+		{
+			name:         "Lambda passed",
+			kind:         "Lambda",
+			expectedKind: ApplicationKind_LAMBDA,
+			valid:        true,
+		},
+		{
+			name:         "ECS passed",
+			kind:         "ECS",
+			expectedKind: ApplicationKind_ECS,
+			valid:        true,
+		},
+		{
+			name:         "Invalid app kind passed",
+			kind:         "Kubernetest",
```

Comment:
Tbh, This took time for me to understand this test case. 😄 
nit: Just using something more readable like "InvalidApp" is fine to me.

Reply:
Let's be always explicit 👍

Reply:
I understand 'Kubernetest" is one of the common mistakes we tend to make though.

Reply:
Gotcha 😄

Reply:
Updated https://github.com/pipe-cd/pipe/pull/2934/commits/3841dcc98356876265450618b4de48a32ce99f7c, PTAL